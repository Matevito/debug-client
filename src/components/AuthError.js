import React from 'react'

export const AuthError = ({ error }) => {
    if (!error) {
        return (
            <>
            </>
        )
    }
    else {
        return (
            <div>
                
            </div>
        )
    }
}
