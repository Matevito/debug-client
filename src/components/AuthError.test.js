import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'

//tested comp
import { AuthError } from "./AuthError";

describe("AuthError component", () => {
    test("handles no prop errors", () => {
        const { container} = render(<AuthError error={null}/>);
        expect(container).toMatchSnapshot();
    });

    test("handles errors props as an array", () => {
        const testErrors = [
            {
                param: "first",
                msg: "1"
            },
            {
                param: "second",
                msg: "2"
            },
            {
                param: "third",
                msg: "3"
            }
        ];
        render(<AuthError error={testErrors} />);
        expect(screen.getAllByRole("alert").length).toBe(3)
        expect(screen.getByText("first error")).toBeInTheDocument();
        expect(screen.getByText("second error")).toBeInTheDocument();
        expect(screen.getByText("third error")).toBeInTheDocument();
        expect(screen.getByText("1")).toBeInTheDocument();
        expect(screen.getByText("2")).toBeInTheDocument();
        expect(screen.getByText("3")).toBeInTheDocument();
    });

    test("handles errors props as a string", () => {
        const errorMsg = "silly mistake!"
        render(<AuthError error={errorMsg} />);

        const displayedError = screen.getByText(errorMsg);
        expect(displayedError).toBeInTheDocument();
    });
})