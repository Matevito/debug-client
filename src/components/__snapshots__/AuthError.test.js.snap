// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthError component handles no prop errors 1`] = `<div />`;
